rp2040-hal/src/timer.rs

```diff
@@ -172,9 +172,9 @@ pub trait Alarm {
     /// reason, the minimum time that this function accepts is `10.microseconds()`
     ///
     /// [enable_interrupt]: #method.enable_interrupt
-    fn schedule<TIME: Into<Microseconds>>(
+    fn schedule<const NOM: u32, const DENOM: u32>(
```

Comment:
Trying to use this I find the ergonomics a bit weird (might be me)...

So in converting I went from: 
`alarm0.schedule(TIME_TICK_MS.microseconds() * 1000);`
to
`alarm0.schedule(TIME_TICK_MS.micros() * 1000);`
which gave me an error because schedule is generic over the input duration NOM and DENOM. Which forces met to do:
`alarm0.schedule::<1, 1000_000>(TIME_TICK_MS.micros() * 1000)`

Since we convert to millis below, does it not make sense to just accept MillisDurationU32 since the caller can easily convert?

Reply:
Mwa maybe not relevant. If I structure my code better this becomes a non issue...

Reply:
I don't understand what you mean.

Doesn't it work if you use `alarm0.schedule(TIME_TICK_MS)` ? (note there's indeed an error with to_micro/to_millis)

Reply:
No in the example TIME_TICKS_MS was an u32. If you make that a proper duration then there is no issue.

However I found that the following is problematic:
`alarm0.schedule(10.micros())` 
You might argue that a constant could be created for that as well. But thats sometimes overkill